go/vt/sqlparser/sql.y

```diff
@@ -2271,6 +2283,136 @@ alter_statement:
     }
   }
 
+partitions_options_opt:
+  {
+    $$ = nil
+  }
+| PARTITION BY linear_opt id_or_var '(' expression ')' partitions_opt
```

Comment:
Why are we using id_or_var here, instead of the keyword `Hash`

Reply:
The test fails(15) if we use the keyword `HASH`, so I used `id_or_var`
```
=== RUN   TestValid/alter_vschema_on_a_add_vindex_hash_(id)
    parse_test.go:2066: Parsing failed. 
        Expected/Got:
        alter vschema on a add vindex hash (id)
        alter vschema on a add vindex `hash` (id)
    --- FAIL: TestValid/alter_vschema_on_a_add_vindex_hash_(id) (0.00s)
```

Reply:
You should change the expected output of the test and keep `Hash`